File: ContractTests/DotNET/ModelBoundCommandValidator.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Arc.Commands;
using FluentValidation;

namespace HttpFixture;

/// <summary>
/// Pins model-bound rule failures, custom state and warning thresholds over HTTP.
/// </summary>
public sealed class ModelBoundCommandValidator : CommandValidator<ModelBoundCommand>
{
    /// <summary>
    /// Registers the fixture validation rules.
    /// </summary>
    public ModelBoundCommandValidator()
    {
        RuleFor(command => command.Title).NotEmpty().WithMessage("Title required").WithState(_ => "title-owned");
        RuleFor(command => command.Title).MinimumLength(5).WithMessage("Consider a longer title").WithSeverity(Severity.Warning);
    }
}

File: ContractTests/Http/modelBound/ModelBoundCommandValidator.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.
import { CommandValidator, Severity, validator } from '@cratis/arc.core';
import { ModelBoundCommand } from './ModelBoundCommand.js';

@validator(ModelBoundCommand)
export class ModelBoundCommandValidator extends CommandValidator<ModelBoundCommand> {
    constructor() {
        super();
        this.ruleFor(command => command.title).notEmpty().withMessage('Title required').withState('title-owned');
        this.ruleFor(command => command.title).minLength(5).withMessage('Consider a longer title').withSeverity(Severity.Warning);
    }
}
